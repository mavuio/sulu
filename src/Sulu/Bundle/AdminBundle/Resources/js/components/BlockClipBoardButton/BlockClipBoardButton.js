// @flow
import React from 'react';
import Icon from '../Icon';
import Popover from '../Popover';
import blockclipboardbuttonStyles from './blockclipboardbutton.scss';
import BlockClipBoardMenu from '../BlockClipBoardMenu';
import {action, computed, observable} from 'mobx';
import {observer} from 'mobx-react';

type Props = {|
    value: Object,
    onPasteBlocks: (items: array<object>) => void
|};


@observer
export default class BlockClipBoardButton<T: string> extends React.Component<Props<T>> {


    setDropdownButtonRef = (ref: ?ElementRef<'button'>) => {
        this.dropdownButtonRef = ref;
    };


    @action handleDropdownToggle = (event) => {
        event.stopPropagation();
        console.log('#log 1021 open', event);
        this.dropdownOpen = !this.dropdownOpen;
    };

    @action handleDropdownClose = (event) => {
        this.dropdownOpen = false;
        console.log('#log 1021 close',event, this.dropdownOpen );
    };

    ignorePopoverBackdropClick = (event) => {
        event.stopPropagation();
    };


    dropdownButtonRef: ?ElementRef<'button'>;
    @observable dropdownOpen = false;


    render() {
        const {value,onPasteBlocks} = this.props;

        return (
            <>
                <button
                className={blockclipboardbuttonStyles.blockclipboardbutton}
                onClick={this.handleDropdownToggle}
                ref={this.setDropdownButtonRef}
                        >
                <Icon name="su-more-horizontal" />
                        </button>
                     <div onClick={this.ignorePopoverBackdropClick}>
                         <Popover
                anchorElement={this.dropdownButtonRef || undefined}
                onClose={this.handleDropdownClose}
                open={this.dropdownOpen}
                        >
                {
                    (setPopoverRef, styles) => (
                        <div ref={setPopoverRef} style={styles}>
                                 <BlockClipBoardMenu  value={value} onPasteBlocks={onPasteBlocks}
/>
                        </div>
                    )
                }
                         </Popover>
                    </div>
            </>
        );
    }
}


