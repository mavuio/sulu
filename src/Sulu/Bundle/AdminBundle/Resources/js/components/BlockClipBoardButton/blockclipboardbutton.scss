@import "../../containers/Application/colors.scss";

.blockclipboardbutton {
    border: none;
    outline: none;
    background: none;
    box-shadow: none;
    cursor: pointer;
    background: $silver;
    padding: 3px;
}
