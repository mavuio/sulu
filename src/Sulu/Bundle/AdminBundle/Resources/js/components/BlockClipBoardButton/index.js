// @flow
import BlockClipBoardButton from './BlockClipBoardButton';

export default BlockClipBoardButton;
