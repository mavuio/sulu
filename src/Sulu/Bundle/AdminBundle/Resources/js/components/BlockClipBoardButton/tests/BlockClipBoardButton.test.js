// @flow
import React from 'react';
import {render} from 'enzyme';
import BlockClipBoardButton from '../BlockClipBoardButton';

test('Render a blockclipboardbutton', () => {
    expect(render(<BlockClipBoardButton>Hello world</BlockClipBoardButton>)).toMatchSnapshot();
});
