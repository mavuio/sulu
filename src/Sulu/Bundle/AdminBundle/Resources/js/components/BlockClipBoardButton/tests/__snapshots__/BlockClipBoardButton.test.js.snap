// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a blockclipboardbutton 1`] = `
<div
  class="blockclipboardbutton"
>
  Hello world
</div>
`;
